File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> map;
        int n = nums.size();
        if (n==0){
            return 0;
        }
        int curr = 0;
        int res = 0;
        map[0] = 1;
        for (int i : nums){
            curr += i;
            if (map.find(curr - k)!= map.end()){
                res += map[curr-k];
            }
            map[curr]++;
        }
        return res;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pref = {0:1}
        res, sm = 0, 0

        for i in nums:
            sm += i
            if (sm-k) in pref:
                res += pref[(sm-k)]
            if sm in pref:
                pref[sm] += 1
            else:
                pref[sm] = 1
        return res